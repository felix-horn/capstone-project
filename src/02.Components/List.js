import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import ListItem from '../01.UI-Elements/ListItem'

List.propTypes = {
  list: PropTypes.object.isRequired,
  listAllocation: PropTypes.array.isRequired,
  toggleIsChecked: PropTypes.func.isRequired,
  checked: PropTypes.string,
}

export default function List({
  list,
  listAllocation,
  toggleIsChecked,
  checked,
}) {
  return (
    <>
      {listAllocation.length > 0 && (
        <ListStyled checked={checked}>
          {listAllocation.map((id) => {
            const { title, isChecked } = list.byId[id]
            return (
              <ListItem
                key={id}
                title={title}
                isChecked={isChecked}
                onToggle={() => toggleIsChecked(id)}
              />
            )
          })}
        </ListStyled>
      )}
    </>
  )
}

const ListStyled = styled.div`
  margin: 3px 2px 10px;
  padding-top: ${(props) => (props.checked ? '10px' : 'none')};
  border-top: ${(props) => (props.checked ? '1px solid #dadcdf' : 'none')};
  display: grid;
`
