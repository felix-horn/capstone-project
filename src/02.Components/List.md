```js padded
<List list={
    {
        allIds: ["02c41e96", "0cb406e2", "7374dca9"],
        byId: {
            "02c41e96": { id: "02c41e96", titleListItem: "Milch", isChecked: false },
            "0cb406e2": { id: "0cb406e2", titleListItem: "Butter", isChecked: false},
            "7374dca9": { id: "7374dca9", titleListItem: "KÃ¤se", isChecked: true },
        }
    }
}/>
```