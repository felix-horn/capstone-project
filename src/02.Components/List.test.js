import { render } from '@testing-library/react'
import List from './List'

describe('List', () => {
  it('renders the list correctly', () => {
    const { container } = render(
      <List
        list={{
          allIds: ['02c41e96', '0cb406e2', '7374dca9'],
          byId: {
            '02c41e96': {
              id: '02c41e96',
              titleListItem: 'Milk',
              isChecked: false,
            },
            '0cb406e2': {
              id: '0cb406e2',
              titleListItem: 'Butter',
              isChecked: false,
            },
            '7374dca9': {
              id: '7374dca9',
              titleListItem: 'Cheese',
              isChecked: true,
            },
          },
        }}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
