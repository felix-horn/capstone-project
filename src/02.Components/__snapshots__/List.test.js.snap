// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders the list correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  margin-right: 10px;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c3 {
  margin-left: 9px;
}

.c0 {
  display: grid;
  gap: 12px;
  margin: 3px 2px 10px;
}

<div
  class="c0"
>
  <label
    class="c1"
  >
    <input
      class="c2"
      type="checkbox"
    />
    <span
      class="c3"
    >
      Milk
    </span>
  </label>
  <label
    class="c1"
  >
    <input
      class="c2"
      type="checkbox"
    />
    <span
      class="c3"
    >
      Butter
    </span>
  </label>
  <label
    class="c1"
  >
    <input
      checked=""
      class="c2"
      type="checkbox"
    />
    <span
      class="c3"
    >
      Cheese
    </span>
  </label>
</div>
`;
